hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/metadata/SCMMetadataStoreMetrics.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.hadoop.metrics2.MetricsInfo;
+import org.apache.hadoop.metrics2.MetricsRecordBuilder;
+import org.apache.hadoop.metrics2.MetricsSource;
+import org.apache.hadoop.metrics2.annotation.Metrics;
+import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.metrics2.lib.MetricsRegistry;
+import org.apache.hadoop.ozone.OzoneConsts;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.apache.hadoop.metrics2.lib.Interns.info;
+
+/**
+ * Class contains metrics related to SCM Metadata Store.
+ */
+@Metrics(about = "SCM Metadata Store Metrics", context = OzoneConsts.OZONE)
+public final class SCMMetadataStoreMetrics implements MetricsSource {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(SCMMetadataStoreMetrics.class);
+
+  public static final String METRICS_SOURCE_NAME =
+      SCMMetadataStoreMetrics.class.getSimpleName();
+
+  private static final MetricsInfo ESTIMATED_KEY_COUNT = info(
+      "EstimatedKeyCount",
+      "Tracked estimated key count of all column families");
+
+  private MetricsRegistry registry;
+  private static SCMMetadataStoreMetrics instance;
+
+  private SCMMetadataStoreImpl scmMetadataStore;
+
+  private Map<String, MetricsInfo> columnFamilyMetrics;
+
+  public SCMMetadataStoreMetrics(SCMMetadataStoreImpl scmMetadataStoreImpl) {
+    this.registry = new MetricsRegistry(METRICS_SOURCE_NAME);
+    this.scmMetadataStore = scmMetadataStoreImpl;
+
+    columnFamilyMetrics = scmMetadataStoreImpl.getTableMap().entrySet()
+        .stream().collect(
+        Collectors.toMap(Map.Entry::getKey, e -> getMetricsInfo(e.getKey())));
+  }
+
+  public static synchronized SCMMetadataStoreMetrics create(SCMMetadataStoreImpl
+      scmMetadataStore) {
+    if (instance != null) {
+      return instance;
+    }
+    instance = DefaultMetricsSystem.instance().register(METRICS_SOURCE_NAME,
+        "SCM Metadata store related metrics",
+        new SCMMetadataStoreMetrics(scmMetadataStore));
+    return instance;
+  }
+
+  @Override
+  public void getMetrics(MetricsCollector collector, boolean all) {
+    MetricsRecordBuilder builder = collector.addRecord(METRICS_SOURCE_NAME);
+
+    Map<String, Long> keyCountMap = new HashMap<>();
+    for (Map.Entry<String, MetricsInfo> entry: columnFamilyMetrics.entrySet()) {
+      long count = 0L;
+      try {
+        count = scmMetadataStore.getTableMap().get(entry.getKey())
+            .getEstimatedKeyCount();
+      } catch (IOException e) {
+        LOG.error("Can not get estimated key count for table {}",
+            entry.getKey(), e);
+      }
+      builder.addGauge(entry.getValue(), count);
+      keyCountMap.put(entry.getKey(), count);
+    }
+
+    Gson gson = new Gson();
+    builder.tag(ESTIMATED_KEY_COUNT, gson.toJson(keyCountMap));
+  }
+
+  public static void unRegister() {
```

Comment:
Does `unRegister` need to be synchronized too, as it can access `instance` while register is running? I'm not sure if it does or not, but might be safer to make it synchronized.

Reply:
Sure, let me add it.